plugins/catalog-react/src/components/EntityTagPicker/EntityTagPicker.tsx

```diff
@@ -49,8 +51,18 @@ const checkedIcon = <CheckBoxIcon fontSize="small" />;
 /** @public */
 export const EntityTagPicker = () => {
   const classes = useStyles();
-  const { updateFilters, backendEntities, filters, queryParameters } =
-    useEntityList();
+  const { updateFilters, filters, queryParameters } = useEntityList();
+
+  const catalogApi = useApi(catalogApiRef);
+  const { value: availableTags } = useAsync(async () => {
+    const facet = 'metadata.tags';
+    const { facets } = await catalogApi.getEntityFacets({
+      facets: [facet],
+      filter: filters.kind?.getCatalogFilters(),
+    });
+
+    return facets[facet].map(({ value }) => value);
+  }, [filters.kind]);
```

Comment:
Can you really have this as the condition though? Let's think about when you want to change the set of available tags. Isn't it whenever ANY other filter has changed essentially? Both all backend filters - which may be more than just `kind` - as well as other filters could be seen as affecting the set of available options in other filters right?

Like, if you start from the full unfiltered catalog page, then you want to show all tags in all entities.

But then if you press "Starred" so that you see only three entities in the table - you want to show only the tags of those three entities as options to select right? Otherwise you'll just have a ton of options that you KNOW will lead to zero hits. I dunno man :)

Reply:
is this a current requirement? It doesn't seem to work like that in master 🤔

Reply:
Maybe it isn't ... even though that's how good filtering pages usually work I think.

But even lacking that, can we really use `kind` hardcoded as the only source of reloads here?

Reply:
Decided that this is just fine for now - since the rest of the logic isn't in a place to challenge this anyway. Let's merge and iterate if needed!